File: SummitPro.SharedKernel/Specification/ISpecification.cs
namespace SummitPro.SharedKernel.Specification;

public interface ISpecification<T>
{
	bool IsSatisfied(T entity);
	ISpecification<T> And(ISpecification<T> other);
	ISpecification<T> Or(ISpecification<T> other);
	ISpecification<T> Not(ISpecification<T> other);
}

File: SummitPro.Web/Scripts/Specification/ISpecification.ts
export interface ISpecification<T> {
    isSatisfiedBy(element: T): boolean;
    and(other: ISpecification<T>): ISpecification<T>;
    or(other: ISpecification<T>): ISpecification<T>;
    not(other: ISpecification<T>): ISpecification<T>;
}
